src/Features/LanguageServer/Protocol/DefaultCapabilitiesProvider.cs

```diff
@@ -31,6 +31,16 @@ internal class DefaultCapabilitiesProvider : ICapabilitiesProvider
                 .ToImmutableArray();
         }
 
+        public void Initialize()
+        {
+            // Force completion providers to resolve in initialize, because it means MEF parts will be loaded,
+            // which we need to do before the GetCapabilities call, as that is on the UI thread.
```

Comment:
The comment didn't make it entirely clear here that our goal of initializing MEF parts off the UI thread to avoid taking the assembly loads there. The "we need to do before..." said we want to, but not why.